bazel-mode.el

```diff
@@ -0,0 +1,131 @@
+	(modify-syntax-entry ?# "<" table)
+	(modify-syntax-entry ?\n ">" table)
+	table))
+
+(defconst bazel-constants
+  '("true"))
+
+(defconst bazel-events
+  '())
+
+;; define several category of keywords
+(defconst bazel-keywords
+  '("for"))
+
+(defun bazel-mode--sort-strings-longest-first (l)
+  "Sort strings by length and put the longest strings first.
+L is the list of strings to sort."
+  (sort l #'(lambda (s1 s2) (> (length s1) (length s2)))))
+
+;; arguments to functions
+(defconst bind-args
+  '("actual"))
+
+(defconst cc-buildrule-args
+  '("copts" "defines" "hdrs" "includes" "include_prefix" "linkopts" "linkshared"
+	"linkstatic" "malloc" "nocopts" "strip_include_prefix" "textual_headers"))
+
+(defconst common-buildrule-args
+  '("deps" "data" "args" "defines" "stamp" "toolchains"))
+
+(defconst common-testrule-args
+  '("local" "flaky" "shard_count" "size" "timeout"))
+
+(defconst config-setting-args
+  '("values"))
+
+(defconst genrule-args
+  '("cmd" "compatible_with" "deprecation" "distribs" "executable" "features"
+	"licenses" "message" "name" "output_licenses" "output_to_bindir" "outs"
+	"restricted_to" "srcs" "tags" "testonly" "tools" "visibility"))
+
+(defconst genquery-args
+  '("deps" "data" "expression" "opts" "scope" "strict"))
+
+(defconst git-repository-args
+  '("commit" "init_submodules" "remote" "sha256"))
+
+(defconst glob-args
+  '("exclude" "exclude_directories" "include"))
+
+(defconst http-args
+  '("executable" "strip_prefix" "type" "url" "urls"))
+
+(defconst java-buildrule-args
+  '("classpath_resources" "create_executable" "deploy_manifest_lines"
+	"javacopts" "jvm_flags" "launcher" "main_class" "output_licenses" "plugins"
+	"resources" "resource_jars" "resource_strip_prefix" "restricted_to"
+	"runtime_deps" "use_testrunner"))
+
+(defconst local-repository-args
+  '("path"))
+
+(defconst maven-args
+  '("artifact" "repository" "server" "sha1"))
+
+(defconst new-repository-args
+  '("build_file" "build_file_content" "workspace_file" "workspace_file_content"))
+
+(defconst package-args
+  '("default_deprecation" "default_testonly" "default_visibility"))
+
+(defconst py-buildrule-args
+  '("default_python_version" "default_runtime" "files" "imports" "interpreter"
+    "interpreter_path" "main" "protoc" "srcs_version"))
+
+(defconst bazel-types
+  (bazel-mode--sort-strings-longest-first
+    (append bind-args cc-buildrule-args common-buildrule-args common-testrule-args
+	  config-setting-args genrule-args genquery-args git-repository-args glob-args
+	  http-args java-buildrule-args local-repository-args maven-args
+	  new-repository-args package-args py-buildrule-args)))
+
+;; generate regex string for each category of keywords
+(defvar bazel-keywords-regexp (regexp-opt bazel-keywords 'words))
+(defvar bazel-type-regexp (regexp-opt bazel-types 'words))
+(defvar bazel-constant-regexp (regexp-opt bazel-constants 'words))
+(defvar bazel-event-regexp (regexp-opt bazel-events 'words))
+
+(defvar bazel-font-lock-keywords
```

Comment:
You can make this a defconst and make it internal (i.e. have the name contain a `--`) because users aren't supposed to modify it.

Reply:
Done.